src/scripts/main.js

```diff
@@ -1,3 +1,79 @@
 'use strict';
 
-// write code here
+const container = document.querySelector('.container');
+const field = document.querySelector('.field');
+const fieldBody = field.tBodies[0];
+
+container.addEventListener('click', handleFieldControls);
+
+function handleFieldControls(e) {
+  const button = e.target;
+
+  if (button.tagName !== 'BUTTON') {
+    return;
+  }
+
+  const isRowControl =
+    button.classList.contains('append-row') ||
+    button.classList.contains('remove-row');
+
+  const isColumnControl =
+    button.classList.contains('append-column') ||
+    button.classList.contains('remove-column');
+
+  if (isRowControl) {
+    handleRows(button);
+  }
+
+  if (isColumnControl) {
+    handleColumns(button);
+  }
+
+  updateButtonStates();
+}
+
+function handleRows(button) {
+  if (button.classList.contains('append-row') && fieldBody.rows.length < 10) {
+    const newRow = fieldBody.rows[0].cloneNode(true);
+
+    fieldBody.append(newRow);
+  }
+
+  if (button.classList.contains('remove-row') && fieldBody.rows.length > 2) {
+    fieldBody.lastElementChild.remove();
+  }
+}
+
+function handleColumns(button) {
+  Array.from(fieldBody.rows).forEach((row) => {
+    if (button.classList.contains('append-column') && row.cells.length < 10) {
+      const newCell = row.cells[0].cloneNode(true);
+
+      row.append(newCell);
+    }
+
+    if (button.classList.contains('remove-column') && row.cells.length > 2) {
+      row.lastElementChild.remove();
+    }
+  });
+}
+
+function updateButtonStates() {
+  const rowCount = fieldBody.rows.length;
+  const columnCount = fieldBody.rows[0]?.cells.length || 0;
```

Comment:
The optional chaining operator `?.` is used here to safely access `cells.length`. Ensure that `fieldBody.rows[0]` is not undefined before this line, or handle the case where there are no rows in `fieldBody`.